### PR TITLE
Extract Android SDK version from named platform dirs.

### PR DESCRIPTION
**Pull request by tvolkert**
Previously, we were mapping certain named platforms
(e.g. `android-stable`) to their corresponding version.
this had two problems:

1. The version could become out of date. For instance, we had
   mapped `android-stable` to version 24, but the stable version
   is now 27.
2. The list of possible named versions wasn't comprehensive.
   Some Android SDKs just list the platform as `stable`, or
   `experimental`, etc.

This change updates the platform version detection to use
the `build.prop` file that exists in the platform directory
(only for cases where the version number is not encoded into
the directory name).

### CODE REVIEW

File: packages/flutter_tools/lib/src/android/android_sdk.dart [hunk 1]
```diff
@@ -142,15 +141,14 @@ class AndroidSdk {
...
 
     final Directory platformsDir = fs.directory(fs.path.join(directory, 'platforms'));
     if (platformsDir.existsSync()) {
       platforms = platformsDir
         .listSync()
-        .map<String>((FileSystemEntity entity) => fs.path.basename(entity.path))
-        .where((String name) => name.startsWith('android-'))
-        .toList();
+        .where((FileSystemEntity entity) => entity is Directory)
+        .map<Directory>((FileSystemEntity e) {final Directory d = e; return d;});
```
Comment (cbracken) [inline packages/flutter_tools/lib/src/android/android_sdk.dart]: This should be broken onto a couple lines... even though I admit I kind of like this.

File: packages/flutter_tools/lib/src/android/android_sdk.dart [hunk 2]
```diff
@@ -209,22 +217,18 @@ class AndroidSdk {
...
   final Version buildToolsVersion;
-  String get buildToolsVersionName => buildToolsVersion.toString();
 
-  int get sdkLevel {
-    if (_namedVersionMap.containsKey(platformVersionName))
-      return _namedVersionMap[platformVersionName];
-    else
-      return int.parse(platformVersionName.substring('android-'.length));
-  }
+  String get buildToolsVersionName => buildToolsVersion.toString();
```
Comment (cbracken) [inline packages/flutter_tools/lib/src/android/android_sdk.dart]: Given that we depend on this being non-null, consider adding asserts in the ctor. Looks like we defend against that on ll. 199-200, but probably worth the extra paranoia.

**Review by cbracken — APPROVED** [review]

**Comment by cbracken** [comment]
lgtm modulo comments!

lgtm

**Comment by cbracken** [comment]
Your patch vs `pub`:

So close...

Re-lgtm.

**Comment by tvolkert** [comment]
puuuub

**Pull request merged**